Makefile

```diff
@@ -3263,7 +3270,7 @@ dist: git-archive$(X) configure
 	@$(MAKE) -C git-gui TARDIR=../.dist-tmp-dir/git-gui dist-version
 	./git-archive --format=tar \
 		$(GIT_ARCHIVE_EXTRA_FILES) \
-		--prefix=$(GIT_TARNAME)/ HEAD^{tree} > $(GIT_TARNAME).tar
+		--prefix=$(GIT_TARNAME)/ HEAD > $(GIT_TARNAME).tar
```

Comment:
likewise the ^{tree} change.

Reply:
Removing the `^{tree}` makes `git archive` evaluate with a commit object rather than a tree, which ultimately lets it set the commit hash as a header in the generated `.tar` file (see the second paragraph [here](https://git-scm.com/docs/git-archive#_description)). That header can then be used to set `GIT_BUILT_FROM_COMMIT` when building from an extracted archive (e.g., in the MacOS build).